api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -275,6 +275,11 @@ message ExternalProcessor {
   // backend stream lifetime. In this case, Envoy will eventually timeout the external processor stream according to this time limit.
   // The default value is 5000 milliseconds (5 seconds) if not specified.
   google.protobuf.Duration deferred_close_timeout = 19;
+
+  // [#not-implemented-hide:]
```

Comment:
Could you please also add the comment about the server limitation like server must send back the header response before the body, whether separately or together.  

Also, comment about what happen if server break this contract e.g.,  spurious message.

Reply:
Is that implicit in the contract already?  i.e, adding this knob does not change that contract?

If we want to add such server limitation, it's probably better to be added in the ProcessResponse API, which we do not change :https://github.com/envoyproxy/envoy/blob/6592faae7203ecba8d00ad6e965f9f1be44aea7a/api/envoy/service/ext_proc/v3/external_processor.proto#L131

Reply:
No, before this pr sever won't even receive the body request from ext_proc filter unless it send back header response.

It is key point/change of this pr and necessary restriction on server. I don't see the downside of call it out explicitly

Reply:
I give it a try like below:
"
 // [#not-implemented-hide:]                                                                                                                      
  // Enables sending body to the side stream server before receives the header response from that server.                                          
  // It only works for STREAMED body processing mode. For any other body processing modes, it is ignored.                                          
  // With this enabled, in case the server chooses to wait for the body before sending back the response,                                          
  // it may include the header response and the 1st chunk body response in one response message. If the                                            
  // server chooses to send the header response separately, that has to be sent first.                                                  
  bool streaming_no_header_resp = 21;
"

Reply:
question: if this field is set, and server chooses to behave as though its today's model (i.e. respond for header immediately, respond for each chunk, etc), what happens in Envoy?

If nothing is impacted in envoy, why even have the flag? we can have runtime behavior to protect new code.

Secondly, even with this flag, what happens if server switches the mode in header response from streamed to buffered, etc? I feel that for m:n streaming, disallow_processing_mode_overrides flag must be set to true to avoid code complexity. Put another way, can we do this:
 always enable m:n streaming if body mode is streaming and processing mode override is false? Its a superset of 1-1 streaming anyway.

Reply:
For your 1st question, I think it should be allowed. I.e, server can choose sending header response without waiting for the body, like today. Or wait for the body then send back responses. In my opinion, as long as the server sends back responses in order, that's fine.

For you 2nd question, the flag is to give the server the capability to see the body before sending the response. That's not supported today.  If we directly changing to send body without wait for the header response without this flag configuration, that might confuse server working in legacy mode, i.e. why it sees body before sending back header response?

For the 3nd question, we probably should block process_mode_override if this flag is enabled, or the future MxN mode.

Reply:
@rshriram  Yes, M*N mode can use this feature as default behavior. On Envoy side, decoupling this from M*N  has following benefits :
(1) It is a beneficial to existing 1:1 streaming mode, with backwards compatibility 
(2) It can address some immediate customers' asks, which also de-risk the M:N support .